### DISCUSSION TITLE
Expand relationship with getFirstListItem

### DISCUSSION THREAD

**Discussion by Graffalo92** [pocketbase/pocketbase#3174] [Q&A] [answered]
Hello , i just want to ask if its possible to retrieve relationship like this 

const data = await pb.collection("collection").getFirstListItem(`value="${slug}"` , {expand: "some" });

I have assign relationship to the main collection and i have also remove api rules but no lack returning expand data. 

Thank you

**Comment by ganigeorgiev** [comment]
The above call format seems correct.

Double check the View API rule of the expanded collection.

If you are not able to make it work, please provide a more specific example and paste your collections configuration as JSON (_you can export it from the Admin UI > Settings > Export collections_).

**Comment by Graffalo92** [comment]
Find the json  of my collection. I have checked and api rules are now apply and are available to anyone. 
    const data = await pb .collection("destinations").getFirstListItem(`route_name="${slug}"` , {expand: "metadata"});
{
    "id": "n6ijgf9xbnic7vu",
    "name": "destinations",
    "type": "base",
    "system": false,
    "schema": [
        {
            "id": "w4femzvm",
            "name": "destination_name",
            "type": "json",
            "system": false,
            "required": false,
            "options": {}
        },
        {
            "id": "u9rcydjc",
            "name": "thumbnail",
            "type": "file",
            "system": false,
            "required": false,
            "options": {
                "maxSelect": 1,
                "maxSize": 5242880,
                "mimeTypes": [],
                "thumbs": [],
                "protected": false
            }
        },
        {
            "id": "3eq9j3ak",
            "name": "carousel",
            "type": "file",
            "system": false,
            "required": false,
            "options": {
                "maxSelect": 99,
                "maxSize": 5242880,
                "mimeTypes": [],
                "thumbs": [],
                "protected": false
            }
        },
        {
            "id": "budp01ue",
            "name": "is_active",
            "type": "bool",
            "system": false,
            "required": false,
            "options": {}
        },
        {
            "id": "aplwlavu",
            "name": "thumbnail_blue_hash",
            "type": "text",
            "system": false,
            "required": false,
            "options": {
                "min": null,
                "max": null,
                "pattern": ""
            }
        },
        {
            "id": "uwxtok41",
            "name": "title",
            "type": "json",
            "system": false,
            "required": false,
            "options": {}
        },
        {
            "id": "dcy5ufq1",
            "name": "text_en",
            "type": "editor",
            "system": false,
            "required": false,
            "options": {}
        },
        {
            "id": "o0lvo5yq",
            "name": "text_es",
            "type": "editor",
            "system": false,
            "required": false,
            "options": {}
        },
        {
            "id": "qlagud3g",
            "name": "text_pt",
            "type": "editor",
            "system": false,
            "required": false,
            "options": {}
        },
        {
            "id": "vq8kiplw",
            "name": "route_name",
            "type": "text",
            "system": false,
            "required": false,
            "options": {
                "min": null,
                "max": null,
                "pattern": ""
            }
        },
        {
            "id": "uevrbkbb",
            "name": "metadata_id",
            "type": "relation",
            "system": false,
            "required": false,
            "options": {
                "collectionId": "loylx4rpymo22xr",
                "cascadeDelete": false,
                "minSelect": null,
                "maxSelect": 1,
                "displayFields": []
            }
        }
    ],
    "indexes": [],
    "listRule": "",
    "viewRule": "",
    "createRule": null,
    "updateRule": null,
    "deleteRule": null,
    "options": {}
}

**Reply by ganigeorgiev** [reply, marked as answer]
The relation field seems to be named `metadata_id`, not `metadata`.

Also the above doesn't show the "metadata" collection, so if the above is not a typo in the `expand` prop, double check the "metadata" collection View API rule.

**Reply by Graffalo92** [reply]
Ok got it. you expand with relation field name. i though it was configured through pocketbase to automatically find the relation. thank you

**Reply by ganigeorgiev** [reply]
Yes, when using direct `expand` you have to use the relation field name. The reason for this is that you can have more than one relation field pointing to the same collection.
